cfme/tests/infrastructure/test_customization_template.py

```diff
@@ -1,101 +1,113 @@
 # -*- coding: utf-8 -*-
 import fauxfactory
 import pytest
-
-from cfme.infrastructure import pxe
 from cfme.utils import error
-from cfme.utils.blockers import BZ
 from cfme.utils.update import update
+from cfme.utils import version
 
 pytestmark = [pytest.mark.tier(3)]
 
 
-def test_customization_template_crud():
+def test_customization_template_crud(appliance):
     """Basic CRUD test for customization templates."""
-    template_crud = pxe.CustomizationTemplate(
-        name=fauxfactory.gen_alphanumeric(8),
-        description=fauxfactory.gen_alphanumeric(16),
-        image_type='RHEL-6',
-        script_type='Kickstart',
-        script_data='Testing the script')
+    collection = appliance.collections.customizationtemplate
+    template_crud = collection.create(name=fauxfactory.gen_alphanumeric(8),
+                                      description=fauxfactory.gen_alphanumeric(16),
+                                      image_type='RHEL-6',
+                                      script_type='Kickstart',
+                                      script_data='Testing the script')
 
-    template_crud.create()
     with update(template_crud):
         template_crud.name = template_crud.name + "_update"
-    template_crud.delete(cancel=False)
+    template_crud.delete()
 
 
-def test_name_required_error_validation():
+def test_name_required_error_validation(appliance):
     """Test to validate name in customization templates."""
-    template_name = pxe.CustomizationTemplate(
-        name=None,
-        description=fauxfactory.gen_alphanumeric(16),
-        image_type='RHEL-6',
-        script_type='Kickstart',
-        script_data='Testing the script')
 
+    collection = appliance.collections.customizationtemplate
     with error.expected('Name is required'):
-        template_name.create()
+        collection.create(
+            name=None,
+            description=fauxfactory.gen_alphanumeric(16),
+            image_type='RHEL-6',
+            script_type='Kickstart',
+            script_data='Testing the script')
 
 
-def test_type_required_error_validation():
+def test_type_required_error_validation(appliance):
     """Test to validate type in customization templates."""
-    template_name = pxe.CustomizationTemplate(
-        name=fauxfactory.gen_alphanumeric(8),
-        description=fauxfactory.gen_alphanumeric(16),
-        image_type='RHEL-6',
-        script_type='<Choose>',
-        script_data='Testing the script')
 
+    collection = appliance.collections.customizationtemplate
     with error.expected('Type is required'):
-        template_name.create()
+        collection.create(
+            name=fauxfactory.gen_alphanumeric(8),
+            description=fauxfactory.gen_alphanumeric(16),
+            image_type='RHEL-6',
+            script_type='<Choose>',
```

Comment:
**REQUIRED:** Though this was how it was done before, I don't like instantiating the class with a ```script_type``` of ```<Choose>``` it's not data, it's just a shim to get it to fill correctly. I far prefer that in the create/update methods we sanitise the dict data and replace None, with ```<Choose>```

Reply:
@psav: It is for error validation test case, where the test doesn't fill any thing i.e selects <Choose> from the dropdown and then does the error validation.

Reply:
Will make the changes.